sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -860,3 +870,30 @@ object PushableColumnAndNestedColumn extends PushableColumnBase {
 object PushableColumnWithoutNestedColumn extends PushableColumnBase {
   override val nestedPredicatePushdownEnabled = false
 }
+
+/**
+ * Get the SQL string of an expression that can be pushed down.
+ */
+object PushableExpression {
```

Comment:
The code size here can go very big very fast. I'd like to have a new class `ExpressionSQLBuilder` in a new file, probably under the package `org.apache.spark.sql.catalyst.utils`, in the `sql/core` module. It can be used like `new ExpressionSQLBuilder(catalystExpr).build()`

Reply:
OK